lib/AST/Module.cpp

```diff
@@ -1191,12 +1191,11 @@ SourceFile::getImportedModules(SmallVectorImpl<ModuleDecl::ImportedModule> &modu
       requiredFilter |= ModuleDecl::ImportFilterKind::ImplementationOnly;
     else if (desc.importOptions.contains(ImportFlags::SPIAccessControl))
       requiredFilter |= ModuleDecl::ImportFilterKind::SPIAccessControl;
+    else if (!separatelyImportedOverlays.lookup(desc.module.importedModule).empty())
+      requiredFilter |= ModuleDecl::ImportFilterKind::ShadowedBySeparateOverlay;
     else
       requiredFilter |= ModuleDecl::ImportFilterKind::Private;
 
-    if (!separatelyImportedOverlays.lookup(desc.module.importedModule).empty())
-      requiredFilter |= ModuleDecl::ImportFilterKind::ShadowedBySeparateOverlay;
-
```

Comment:
This "fix" is incorrect. The `ShadowedBySeparateOverlay` filter is essentially a different axis altogether.

```
             | @_exported | private | @_implementationOnly | @_spi
-------------+------------+---------+----------------------+------
Not shadowed |     EN     |   PN    |        IN            |  SN
Shadowed     |     ES     |   PS    |        IS            |  SS
```

Ideally, we'd be passing in two `OptionSet`s, not one IMO. If you pass in `newFilter = {{@_exported}, {not shadowed}}`, you get EN. If you pass in `newFilter = {{@_exported | private}, {shadowed}}`, you'd get ES and PS.

Today, if you pass in `filter = {@_exported}`, you get EN whereas if you pass in `filter = {shadowed}`, you get nothing. That's unnecessarily confusing.